helm-mt.el

```diff
@@ -73,24 +82,45 @@
         (action . (("Switch to terminal buffer" . (lambda (candidate)
                                                     (switch-to-buffer candidate)))
                    ("Exit marked terminals" 'helm-mt/delete-marked-terms)))))
- 
 
-(defvar helm-mt/term-source-terminal-not-found
-  '((name . "Launch a new terminal")
-    (dummy)
-    (action . (("Launch new terminal" . (lambda (candidate)
-                                          (helm-mt/launch-term candidate)))))))
+(defun helm-mt/term-source-terminal-not-found ()
+  `((name . "Launch a new terminal")
+	(dummy)
+	(action . ,(mapcar (lambda (mode)
+						  `(,(format "Launch new %s" mode) . 
+							(lambda (candidate)
+								(helm-mt/launch-term candidate (quote ,mode)))))
+					  helm-mt/all-terminal-modes))))
+
+(defun helm-mt/shell-advice (orig-fun &rest args)
+  (message "wrapping shell with helm-mt")
+  (if (called-interactively-p 'interactive)
+	  (call-interactively 'helm-mt)
+	(apply orig-fun args)))
+
+;;;###autoload
+(defun helm-mt/wrap-shells (onoff)
+  "Put advice around shell functions when called interactively that routes to helm-mt UI instead of launching a new shell/terminal. If ONOFF is t, activate the advice and if nil, remove it."
+  (interactive)
+  (dolist (mode helm-mt/all-terminal-modes)
+	(let ((fun (intern (string-replace-2 "-mode" "" (symbol-name mode)))))
```

Comment:
@cwhatley do you know where this function `string-replace-2` comes from? Doesn't seem to be a built-in function